src/storage/core.rs

```diff
@@ -304,7 +304,12 @@ where
         if self.try_create_active_blob().await.is_ok() {
             info!("Active blob was set during write operation");
         }
-        if !self.inner.config.allow_duplicates() && self.contains_with(key, meta.as_ref()).await? {
+        if !self.inner.config.allow_duplicates()
+            && matches!(
```

Comment:
It is better to declare additional functions on `ReadResult<>` and use them instead of `match!` macro